### PR TITLE
RD-6372 create Secret Provider column in Secret widget

### PR DESCRIPTION
**Pull request by PatrycjaKubiczek**
## Description

RD-6372
- add a new column for Secret Provider
- change name from `hidden value` to `hidden`
- adjust columns widths

## Screenshots / Videos

Zrzut ekranu 2023-01-04 140432

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

 

n/a

## Documentation

I'll update the documentation after all changes about secret widget will be finished

[RD-1234]: https://cloudifysource.atlassian.net/browse/RD-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RD-6372]: https://cloudifysource.atlassian.net/browse/RD-6372?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: widgets/secrets/src/SecretsTable.tsx [hunk 1]
```diff
@@ -258,6 +259,7 @@ export default class SecretsTable extends React.Component<SecretsTableProps, Sec
                                 </DataTable.Data>
                                 <DataTable.Data>{item.created_at}</DataTable.Data>
                                 <DataTable.Data>{item.updated_at}</DataTable.Data>
+                                <DataTable.Data>{item.provider_name}</DataTable.Data>
```
Comment (qooban) [inline widgets/secrets/src/SecretsTable.tsx:262]: Just out of curiosity how provider name looks like?

I'm just wondering should we format it somehow (e.g. letter case changes). Not necessarily within this PR.

File: widgets/secrets/src/SecretsTable.tsx [hunk 2]
```diff
@@ -211,16 +211,17 @@ export default class SecretsTable extends React.Component<SecretsTableProps, Sec
...
+                        label={translateSecrets('columns.secretProvider')}
+                        name="provider_name"
                         width="10%"
                     />
-                    <DataTable.Column label={translateSecrets('columns.created')} name="created_at" width="10%" />
-                    <DataTable.Column label={translateSecrets('columns.updated')} name="updated_at" width="10%" />
                     <DataTable.Column label={translateSecrets('columns.creator')} name="created_by" width="10%" />
                     <DataTable.Column label={translateSecrets('columns.tenant')} name="tenant_name" width="10%" />
-                    <DataTable.Column width="10%" />
+                    <DataTable.Column width="5" />
```
[the remark this answers is not in the document]
Comment (PatrycjaKubiczek) [reply widgets/secrets/src/SecretsTable.tsx]: 5% indeed! thanks, added it here: https://github.com/cloudify-cosmo/cloudify-stage/pull/2359/commits/7db417b2356b89408ed86431e540b89f2dc7cb16

File: widgets/secrets/src/SecretsTable.tsx (hunk 1 above)
Comment (PatrycjaKubiczek) [reply widgets/secrets/src/SecretsTable.tsx:262]: hmm from what I remember we have this requirements about secret provider name:  `Only letters, digits, and the characters "-", "." and "_" are allowed.`

Comment (qooban) [reply widgets/secrets/src/SecretsTable.tsx:262]: No, no, I meant what are the actual names :-) 
Probably something like "local", "vault", ... ?

**Review by qooban — APPROVED** [review]

**Review by kubama — COMMENTED** [review]
Can we extend existing system test to cover the new feature?

**Comment by PatrycjaKubiczek** [comment]
@kubama could I prepare a system test in the upcoming PR about this functionality? (https://cloudifysource.atlassian.net/browse/RD-6370) here I'll add real secret provider names to the secret widget table. Right now it's only a blank column so I don't see the purpose of writing a test to only check if this column exists. WDYT?

File: widgets/secrets/src/SecretsTable.tsx (hunk 1 above)
Comment (PatrycjaKubiczek) [reply widgets/secrets/src/SecretsTable.tsx:262]: well, I meant this is what secret provider **_names_** looks like, "vault" and "local" is a provider **_type_** 🤔 see  [docs api](https://docs.cloudify.co/api/staging/dev/#get-secrets-provider). Or am I missing something?

**Comment by kubama** [comment]
> Right now it's only a blank column

I'm not sure I understand, from what I can see the column is populated with `item.provider_name`.

Of course we can add system test in upcoming PR, but if so it would be good to at least verify if existing secrets widget test is not broken due to the changes.

File: widgets/secrets/src/SecretsTable.tsx (hunk 1 above)
Comment (qooban) [reply widgets/secrets/src/SecretsTable.tsx:262]: Right, my bad. I mixed name with type. All good then, let's see how it works after RD-6370.

**Comment by qooban** [comment]
Regarding tests:

1. I think it's OK to add it once we have RD-6370 ready.
2. Running `widgets/secrets_spec` prior merging this PR is of course strongly recommended. I would also consider running `flows/user_spec` as it's playing with secrets as well.

**Review by Vorbert-Kruk — APPROVED** [review]
LGTM 🚀 
Regarding tests subject, I have a similar opinion as [qooban](https://github.com/cloudify-cosmo/cloudify-stage/pull/2359#issuecomment-1371839453)

**Comment by PatrycjaKubiczek** [comment]
I ran the system test (secrets_spec and user_spec) https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/4238/ - there is failed test on user_spec but it's a known issue

**Review by kubama — APPROVED** [review]

**Pull request merged**